src/overlays/actors/ovl_En_M_Thunder/z_en_m_thunder.c

```diff
@@ -29,39 +39,533 @@ const ActorInit En_M_Thunder_InitVars = {
+                AudioSfx_PlaySfx(NA_SE_IT_SWORD_SWING_HARD, &player->actor.projectedPos, 4, &gSfxDefaultFreqAndVolScale,
+                                 &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+            }
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+
+        player->stateFlags2 &= ~PLAYER_STATE2_20000;
+
+        if (ENMTHUNDER_GET_MAGIC_COST(&this->actor) != 0) {
+            gSaveContext.magicState = MAGIC_STATE_CONSUME_SETUP;
+        }
+
+        if (player->unk_B08[0] < 0.85f) {
+            this->collider.info.toucher.damage = sDamages[this->type];
+            this->subtype = ENMTHUNDER_SUBTYPE_SPIN_REGULAR;
+            if (this->type == ENMTHUNDER_TYPE_GREAT_FAIRY_SWORD) {
+                this->scaleTarget = 4;
+            } else if (this->type == ENMTHUNDER_TYPE_GILDED_SWORD) {
+                this->scaleTarget = 3;
+            } else {
+                this->scaleTarget = 2;
+            }
+        } else {
+            this->collider.info.toucher.damage = sDamages[this->type + 4];
+            this->subtype = ENMTHUNDER_SUBTYPE_SPIN_GREAT;
+            if (this->type == ENMTHUNDER_TYPE_GREAT_FAIRY_SWORD) {
+                this->scaleTarget = 6;
+            } else if (this->type == ENMTHUNDER_TYPE_GILDED_SWORD) {
+                this->scaleTarget = 4;
+            } else {
+                this->scaleTarget = 3;
+            }
+        }
+
+        if (player->meleeWeaponAnimation < PLAYER_MWA_SPIN_ATTACK_1H) {
+            this->subtype += ENMTHUNDER_SUBTYPE_SWORDBEAM_GREAT;
+            this->actionFunc = EnMThunder_SwordBeam_Attack;
+            this->timer = 1;
+        } else {
+            this->actionFunc = EnMThunder_Spin_Attack;
+            this->timer = 8;
+        }
+
+        AudioSfx_PlaySfx(sChargingSfxIds[this->subtype], &player->actor.projectedPos, 4, &gSfxDefaultFreqAndVolScale,
+                         &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+
+        this->lightColorFrac = 1.0f;
+
+        return;
+    }
+
+    if (!(player->stateFlags1 & PLAYER_STATE1_1000)) {
+        if (this->actor.child != NULL) {
+            this->actor.child->parent = NULL;
+        }
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    if (player->unk_B08[0] > 0.15f) {
+        this->chargingAlpha = 255;
+        if (this->actor.child == NULL) {
+            Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_EFF_DUST, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, this->actor.shape.rot.y, 0,
+                               EFF_DUST_TYPE_2);
+        }
+        this->unk1B4 += (((player->unk_B08[0] - 0.15f) * 1.5f) - this->unk1B4) * 0.5f;
+    } else if (player->unk_B08[0] > .1f) {
+        this->chargingAlpha = (s32)((player->unk_B08[0] - .1f) * 255.0f * 20.0f);
+        this->lightColorFrac = (player->unk_B08[0] - .1f) * 10.0f;
```

Comment:
```suggestion
    } else if (player->unk_B08[0] > 0.1f) {
        this->chargingAlpha = (s32)((player->unk_B08[0] - 0.1f) * 255.0f * 20.0f);
        this->lightColorFrac = (player->unk_B08[0] - 0.1f) * 10.0f;
```
Presumably the zero is dropped for matching?

Reply:
I think a comment saying the missing zeroes are required would be nice